src/OpenTelemetry.ResourceDetectors.Container/ContainerResourceDetector.cs

```diff
@@ -132,45 +152,129 @@ private static string RemovePrefixAndSuffixIfNeeded(string input, int startIndex
         return input.Substring(startIndex, endIndex - startIndex);
     }
 
+    private static string? ExtractK8sContainerId()
+    {
+        try
+        {
+            var host = Environment.GetEnvironmentVariable(K8sServiceHostKey);
+            var port = Environment.GetEnvironmentVariable(K8sServicePortKey);
+            var @namespace = Environment.GetEnvironmentVariable(K8sNamespaceKey);
+            var hostname = Environment.GetEnvironmentVariable(K8sHostnameKey);
+            var containerName = Environment.GetEnvironmentVariable(K8sContainerNameKey);
+            var url = $"https://{host}:{port}/api/v1/namespaces/{@namespace}/pods/{hostname}";
+            var credentials = GetK8sCredentials(K8sCredentialPath);
+            if (string.IsNullOrEmpty(credentials))
+            {
+                return string.Empty;
+            }
+
+            using var httpClientHandler = ServerCertificateValidationHandler.Create(K8sCertificatePath, ContainerResourceEventSource.Log);
+            var response = ResourceDetectorUtils.SendOutRequest(url, "GET", new KeyValuePair<string, string>("Authorization", credentials), httpClientHandler).GetAwaiter().GetResult();
+            var pod = DeserializeK8sResponse(response);
+            if (pod == null || pod.Status == null || pod.Status.ContainerStatuses == null)
+            {
+                return string.Empty;
+            }
+
+            var container = pod.Status.ContainerStatuses.SingleOrDefault(p => p.Name == containerName);
+            if (container is null || string.IsNullOrEmpty(container.Id))
+            {
+                return string.Empty;
+            }
+
+            // Container's ID is in <type>://<container_id> format.
+            var index = container.Id.LastIndexOf('/');
+            return container.Id.Substring(index + 1);
+        }
+        catch (Exception ex)
+        {
+            ContainerResourceEventSource.Log.ExtractResourceAttributesException($"{nameof(ContainerResourceDetector)}: Failed to extract container id", ex);
+        }
+
+        return null;
+
+        static string? GetK8sCredentials(string path)
+        {
+            try
+            {
+                var stringBuilder = new StringBuilder();
+
+                using (var streamReader = ResourceDetectorUtils.GetStreamReader(path))
+                {
+                    while (!streamReader.EndOfStream)
+                    {
+                        stringBuilder.Append(streamReader.ReadLine()?.Trim());
+                    }
+                }
+
+                stringBuilder.Insert(0, "Bearer ");
```

Comment:
Instead of appending the "Bearer " prefix after the fact, could the StringBuilder instance be initialized with the value?

For example:
`var stringBuilder = new StringBuilder("Bearer ");`

Reply:
I reused the logic from AWSEKSResourceDetector
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/9d4b5f102269f883f15974c35e0676aa2186779d/src/OpenTelemetry.ResourceDetectors.AWS/AWSEKSResourceDetector.cs#L80
I changed it in both places. Thank you!